pydarn/plotting/fan.py

```diff
@@ -0,0 +1,248 @@
+# Copyright (C) 2020 SuperDARN Canada, University of Saskatchewan
+# Author: Daniel Billett, Marina Schmidt
+"""
+Fan plots, mapped to AACGM coordinates
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+import aacgmv2
+import datetime as dt
+import os
+
+from pydarn import PyDARNColormaps
+from typing import List
+from matplotlib import ticker, cm, colors
+
+class fan():
+    # TODO: Add class documentation
+
+    def __str__(self):
+        return "This class is static class that provides"\
+                " the following methods: \n"\
+                "   - plot_fan()\n"\
+                "   - return_beam_pos()\n"
+
+    @classmethod
+    def plot_radar_fov(cls, dmap_data: List[dict], scan_index: int = 0,
+                       all_gates: bool = False, parameter: str = 'v',
+                       hemisphere: str = 'north', lowlat: int = 50,
+                       cmap: str = None, groundscatter: bool = False,
+                       zmin: int = None, zmax: int = None,
+                       colorbar: plt.colorbar = None,
+                       colorbar_label: str = ''):
+        # TODO: pyDARN can work out the hemisphere via superDARNRadars.radars
+        # check out the file. Even has an enum for hemisphere and associated
+        # value you will find in MAP files
+        """
+        Plots a radar FOV fan plot for the given data and scan number
+
+        Parameters
+        -----------
+            dmap_data: List[dict]
+                Named list of dictionaries obtained from SDarn_read
+            scan_index: int
+                Scan number from beginning of first record in file
+                Default: 0
+            parameter: str
+                Key name indicating which parameter to plot.
+                Default: v (Velocity). Alternatives: 'p_l', 'w_l', 'elv'
+            hemisphere: str
+                North or south, depending on radar. Currently,
+                pyDARN cannot work this out
+                itself, so you should always give it
+                Default: 'north'. Alternatives: 'south'
+            lowlat: int
+                Lower AACGM latitude boundary for the polar plot
+                Default: 50
+            all_gates: bool
```

Comment:
Could we make this an int and call it `max_gates`? This will allow for more flexibility.

Reply:
Potentially, but it would leave the user to guess the maximum number of gates. Could impliment it so that if they choose a number of max gates, it defaults to max gates?

Reply:
yea I think that would be better. I do that with RTP plots.

Reply:
But what about if I only want E-region fans?  How does a user specify a custom number of range gates?  Also, I frequently plot fans for 1/2F ranges only, which means I need to specify both a custom minimum and maximum.  It doesn't look like I can do that here.

Reply:
Updated functionality. `all_gates` has been removed and replaced with `ranges`, which is a two element list of the lower and upper ranges to plot. Default is [0,75]